pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_unique failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_unique [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8458526?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8458526?buildTab=artifacts#/) on master @ [9caf75810ff7dbfa42d16b746c21c86bfa0c6324](https://github.com/cockroachdb/cockroach/commits/9caf75810ff7dbfa42d16b746c21c86bfa0c6324):


```
=== RUN   TestLogic_unique
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/0205c25cd2c9009394cc81369c4dd2f5/logTestLogic_unique531191922
    test_log_scope.go:79: use -show-logs to present logs inline
[10:40:09] setting distsql_workmem='56671B';
=== CONT  TestLogic_unique
    logic.go:4201: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0205c25cd2c9009394cc81369c4dd2f5/logTestLogic_unique531191922
--- FAIL: TestLogic_unique (2.84s)
=== RUN   TestLogic_unique/Upsert
    logic.go:2576: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1548/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/unique:835: 
        expected:
        pq: duplicate key value violates unique constraint "unique_d"\nDETAIL: Key \(d\)=\(2\.00\) already exists\.
        got:
        pq: duplicate key value violates unique constraint "uniq_computed_pk_c_s_s_key"
        DETAIL: Key (c_s,s)=('a','a') already exists.
    logic.go:2201: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1548/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/unique:838: too many errors encountered, skipping the rest of the input
[10:40:10] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1548/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/unique with config fakedist-disk: 146 tests, 2 failures
[10:40:11] --- total progress: 146 statements
--- total: 146 tests, 2 failures
    --- FAIL: TestLogic_unique/Upsert (0.31s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_unique.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23863

--- a/pkg/sql/logictest/testdata/logic_test/unique
+++ b/pkg/sql/logictest/testdata/logic_test/unique
@@ -829,10 +829,10 @@ statement error pgcode 23505 pq: duplicate key value violates unique constraint 
 INSERT INTO uniq_computed_pk (i, s, d) VALUES (1, 'a', 1.0) ON CONFLICT (s) DO UPDATE SET i = 2
 
 statement error pgcode 23505 pq: duplicate key value violates unique constraint "uniq_computed_pk_c_s_s_key"\nDETAIL: Key \(c_s,s\)=\('b','b'\) already exists\.
-UPSERT INTO uniq_computed_pk (i, s, d) VALUES (1, 'b', 1.0)
+UPSERT INTO uniq_computed_pk (i, s, d) VALUES (3, 'b', 3.0)
 
 statement error pgcode 23505 pq: duplicate key value violates unique constraint "unique_d"\nDETAIL: Key \(d\)=\(2\.00\) already exists\.
-UPSERT INTO uniq_computed_pk (i, s, d) VALUES (1, 'a', 2.00)
+UPSERT INTO uniq_computed_pk (i, s, d) VALUES (3, 'c', 2.00)
 
 query ITFTTFT colnames,rowsort
 SELECT * FROM uniq_computed_pk

